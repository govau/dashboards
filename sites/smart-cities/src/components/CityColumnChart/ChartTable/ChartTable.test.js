import React from 'react';
import { shallow } from 'enzyme';
import ChartTable from './ChartTable';

jest.mock('../../../constants', () => ({
  INDICATORS: {
    oneThing: {
      format: '0.0%',
      legendText: 'One thing',
    },
    anotherThing: {
      format: '0.0%',
      legendText: 'Another thing',
    },
  },
}));

const defaultProps = {
  chart: {
    name: 'The things',
    indicatorIds: [
      'oneThing',
      'anotherThing',
    ],
    description: 'This shows one thing and the other thing.',
  },
  cities: [
    {
      id: 'sydney',
      name: 'Sydney',
      indicators: {
        oneThing: 1,
        anotherThing: 0.62,
      },
    },
    {
      id: 'perth',
      name: 'Perth',
      indicators: {
        oneThing: 0.8,
        anotherThing: 10,
      },
    },
  ],
};

it('should match Snapshot', () => {
  const component = shallow(<ChartTable {...defaultProps} />);
  expect(component.debug()).toMatchSnapshot();
});

it('should render a table', () => {
  const component = shallow(<ChartTable {...defaultProps} />);

  expect(component.find('caption').text()).toBe('Table showing The things for each city. This shows one thing and the other thing.')
});

it('should render each city as a row', () => {
  const component = shallow(<ChartTable {...defaultProps} />);

  expect(component.find('tbody tr').length).toBe(2);
  // test the two values for the Sydney row
  expect(component.find('tbody tr').at(0).find('td').at(0).text()).toBe('100.0%');
  expect(component.find('tbody tr').at(0).find('td').at(1).text()).toBe('62.0%');
});

it('should render each indicator as a col', () => {
  const component = shallow(<ChartTable {...defaultProps} />);

  expect(component.find('thead th').length).toBe(3);
  expect(component.find('thead th').at(1).text()).toBe('One thing');
  expect(component.find('thead th').at(2).text()).toBe('Another thing');
});
