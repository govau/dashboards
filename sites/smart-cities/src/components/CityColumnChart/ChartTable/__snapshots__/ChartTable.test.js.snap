// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<table className={[undefined]}>
  <caption>
    Table showing 
    The things
     for each city. 
    This shows one thing and the other thing.
  </caption>
  <thead>
    <tr>
      <th scope=\\"col\\">
        City
      </th>
      <th scope=\\"col\\">
        One thing
      </th>
      <th scope=\\"col\\">
        Another thing
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th scope=\\"row\\">
        Sydney
      </th>
      <td>
        100.0%
      </td>
      <td>
        62.0%
      </td>
    </tr>
    <tr>
      <th scope=\\"row\\">
        Perth
      </th>
      <td>
        80.0%
      </td>
      <td>
        1000.0%
      </td>
    </tr>
  </tbody>
</table>"
`;
